src/main/java/teammates/ui/controller/StudentCourseJoinAuthenticatedAction.java

```diff
@@ -24,8 +25,16 @@
     @Override
     protected ActionResult execute() throws EntityDoesNotExistException {
         Assumption.assertNotNull(regkey);
```

Comment:
Not necessary. You can see that the change affected only `StudentCourseJoinAction` in which the next URL is very specific, `Const.ActionURIs.STUDENT_COURSE_JOIN_AUTHENTICATED`.
Until next URLs everywhere else is handled the way it is handled here, only this class needs a special treatment.